bigquery_etl/shredder/delete.py

```diff
@@ -27,6 +31,13 @@
     help="Do not make changes, only log actions that would be taken; "
     "use with --log-level=DEBUG to log query contents",
 )
+parser.add_argument(
+    "--read_only",
+    "--read-only",
+    action="store_true",
+    help="Use SELECT * FROM instead of DELETE with dry run queries to prevent errors "
+    "due to read-only permissions being insufficient to dry run DELETE dml",
+)
```

Comment:
re-r? @jklukas this was added after you reviewed, the change is isolated to commit [a362c82](https://github.com/mozilla/bigquery-etl/pull/788/commits/a362c82af02858c79ebeedd602c372194fb4f1d0)